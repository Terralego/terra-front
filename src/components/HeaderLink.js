import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

export const HeaderLink = ({ href, children, ...props }) => {
  if (!href) return <span {...props}>{children}</span>;

  if (href.match(/^http/)) return <a className="header-link" href={href} {...props}>{children}</a>;

  return (
    <NavLink to={href} {...props}>{children}</NavLink>
  );
};

HeaderLink.propTypes = {
  children: PropTypes.element.isRequired,
  href: PropTypes.string,
};

HeaderLink.defaultProps = {
  href: '',
};
export default HeaderLink;
