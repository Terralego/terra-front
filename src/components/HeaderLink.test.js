import React from 'react';
import renderer from 'react-test-renderer';
import { HeaderLink } from './HeaderLink';

jest.mock('react-router-dom', () => ({
  NavLink: ({ children, ...props }) => <a {...props}>{children}</a>,
}));

it('should render correctly with no href', () => {
  const tree = renderer.create(
    <HeaderLink>
      <span>Test</span>
    </HeaderLink>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it('should render correctly with http link', () => {
  const tree = renderer.create(
    <HeaderLink href="https://foo.com">
      <span>Foo</span>
    </HeaderLink>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly with navigation', () => {
  const tree = renderer.create(
    <HeaderLink href="foo">
      <span>Bar</span>
    </HeaderLink>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
