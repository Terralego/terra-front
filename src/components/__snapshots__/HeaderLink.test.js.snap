// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with http link 1`] = `
<a
  className="header-link"
  href="https://foo.com"
>
  <span>
    Foo
  </span>
</a>
`;

exports[`should render correctly with navigation 1`] = `
<a
  to="foo"
>
  <span>
    Bar
  </span>
</a>
`;

exports[`should render correctly with no href 1`] = `
<span>
  <span>
    Test
  </span>
</span>
`;
